var/spack/repos/builtin/packages/simmetrix-simmodsuite/package.py

```diff
@@ -0,0 +1,336 @@
+            spack_env.append_path('CMAKE_PREFIX_PATH', archlib)
+            simmetrix_setKernelCMakePrefixPath(self.spec, archlib, spack_env)
+        elif self.spec.satisfies('sim_os=rhel6'):
+            oslib = simmetrix_getLibDir('rhel6')
+            archlib = join_path(prefix.lib, oslib)
+            spack_env.append_path('CMAKE_PREFIX_PATH', archlib)
+            simmetrix_setKernelCMakePrefixPath(self.spec, archlib, spack_env)
+
+    def setup_environment(self, spack_env, run_env):
+        if self.spec.satisfies('sim_os=rhel7'):
+            oslib = simmetrix_getLibDir('rhel7')
+            archlib = join_path(prefix.lib, oslib)
+            run_env.append_path('CMAKE_PREFIX_PATH', archlib)
+            simmetrix_setKernelCMakePrefixPath(self.spec, archlib, run_env)
+        elif self.spec.satisfies('sim_os=rhel6'):
+            oslib = simmetrix_getLibDir('rhel6')
+            archlib = join_path(prefix.lib, oslib)
+            run_env.append_path('CMAKE_PREFIX_PATH', archlib)
+            simmetrix_setKernelCMakePrefixPath(self.spec, archlib, run_env)
+
+    def install(self, spec, prefix):
+        source_path = self.stage.source_path
+        for release in simmodsuite_releases():
+            simVersion = release['version']
+            if simVersion != spec.version.string:
+                continue
+            for name, atts in release['components'].items():
+                feature = atts[1]
+                if '+' + feature in spec:
+                    if name is 'mscore':
+                        simmetrix_copytree(
+                            join_path(source_path, 'lib'),
+                            prefix.lib)
+                        simmetrix_copytree(
+                            join_path(source_path, 'include'),
+                            prefix.include)
+                    else:
+                        path = join_path(
+                            source_path,
+                            name,
+                            self.version.string)
+                        simmetrix_copytree(path, prefix)
+            for name, atts in release['docs'].items():
+                feature = atts[1]
+                if '+' + feature in spec:
+                    path = join_path(
+                        source_path,
+                        name,
+                        self.version.string)
+                    simmetrix_copytree(path, prefix)
+
+        workdir = join_path(prefix, 'code', 'PartitionWrapper')
+        if '+parallelmesh' in spec:
+            with working_dir(workdir):
+                mpi_id = spec['mpi'].name + spec['mpi'].version.string
+                # build the wrapper lib
+                make("-f", "Makefile.custom",
+                     "CC=%s" % spec['mpi'].mpicc,
+                     "CXX=%s" % spec['mpi'].mpicxx,
+                     "PARALLEL=%s" % mpi_id,
+                     "PQUAL=-%s" % mpi_id,
+                     "OPTFLAGS=-O2 -DNDEBUG -fPIC")
```

Comment:
`-fPIC` only works for some compilers, you should use `self.compiler.pic_flag` to make it work for all compilers.

Reply:
`self.compiler.pic_flag` is being used now.